Gems/Multiplayer/Code/Source/Components/LocalPredictionPlayerInputComponent.cpp

```diff
@@ -128,10 +130,15 @@ namespace Multiplayer
 
     LocalPredictionPlayerInputComponentController::LocalPredictionPlayerInputComponentController(LocalPredictionPlayerInputComponent& parent)
         : LocalPredictionPlayerInputComponentControllerBase(parent)
-        , m_autonomousUpdateEvent([this]() { UpdateAutonomous(m_autonomousUpdateEvent.TimeInQueueMs()); }, AZ::Name("AutonomousUpdate Event"))
+#if AZ_TRAIT_SERVER_ENABLED
         , m_updateBankedTimeEvent([this]() { UpdateBankedTime(m_updateBankedTimeEvent.TimeInQueueMs()); }, AZ::Name("BankTimeUpdate Event"))
+#endif
+
+#if AZ_TRAIT_CLIENT_ENABLED
+        , m_autonomousUpdateEvent([this]() { UpdateAutonomous(m_autonomousUpdateEvent.TimeInQueueMs()); }, AZ::Name("AutonomousUpdate Event"))
```

Comment:
I'm not the biggest fan of this. I think it does a number on the readability but it does separate the logic nicely. An alternative here is defining UpdateBankedTime and UpdateAutonomous as empty on their unimplemented platforms vs undeclared and keeping the events in.

This could potentially be an option at the CodeGen level.